### ISSUE TITLE
[1.01.0022] installer detects if java is installed on the machine to and notify user if not

### ISSUE THREAD

**Issue by zadmarbella** [blackberry/VSPlugin#321]
user needs information on which version of java that has to be installed for the update manager to work.
prequisite: have no java installed
1. install the plugin
2. from vs access Blackberry > Settings

Current behaviour:
![missingjrejdk](https://f.cloud.github.com/assets/1478549/1011064/6ca5790c-0b55-11e3-8f13-125c7f20456b.png)

Workaround:
Add this following path to your PATH variable.
C:\bbndk_vs\features\com.qnx.tools.jre.win32_1.6.0.43\jre\bin

**Comment by dbrgss** [comment]
Emailed Doug Schaeffer to see if they can bundle a JRE with the tool.

**Comment by dbrgss** [comment]
Can you retry this...  I think it might have been fixed resulting from a different bug.

1.01.0025

**Comment by zadmarbella** [comment]
VS2012 Win 8
[1.01.0026] Still reproducible 
[1.01;0027] testing is blocked by #369 

Signing uses java.

**Comment by dbrgss** [comment]
Fixed in 1.01.0030

**Comment by zadmarbella** [comment]
reproducible 1.01.0031 reopening.
filed #400

**Comment by zadmarbella** [comment]
reproducible on build [1.01.0032] on win7 vs2010 and win8 2012

**Comment by zadmarbella** [comment]
reproducible on build [1.01.0035] on win7 vs2010 and win8 2012
it looks to only work the first run of vs subsequent reproduces the issue

**Comment by srinivas4912** [comment]
It is verified and found as fixed using VS2012/VS2013 from build 2.0.0018 on Win 8.1 and Win 7 (64&32) bit.

**Comment by srinivas4912** [comment]
Pre-req: Have no JAVA installed.
It is still reproduced using VS2013 plugin from build 2.0.0018 on Win8.1 and not reproduced using VS2012 on Win7 (64&32)bit.

**Comment by srinivas4912** [comment]
It is reproduced on Win8.1 fresh machine.
Pre-req: 1. Have no JAVA installed
              2. Installed VS2013 plugin from build 2.0.0022

Steps to Reproduce:
1. Open Visual Studio 2013
2. Blackberry > Settings [When we click on Settings it is opening below dialog]
![image](https://f.cloud.github.com/assets/5992406/2136574/ce7c02e4-9314-11e3-8cbc-d2c015e1a7a3.png)

**Comment by srinivas4912** [comment]
It is reproducible on Win7 64 bit for the first time and on fresh machine while using VS2012 plugin from 2.0.0022.

**Comment by dbrgss** [comment]
Can you please verify the following for me.   In you Windows Environment Variables please  check to see if the PATH variable has a path set to the following:  

C:\bbndk_vs\features\com.qnx.tools.jre.win32_1.6.0.43\jre\bin;

Also check to make sure that the path above exists on your computer and that in that directory there is a copy of javaw.exe

**Comment by srinivas4912** [comment]
Verified on both Win8.1 and Win7 64 bit machines for:
1. PATH exist in environment variables.
2. Javaw.exe is also exist on drive C:\bbndk_vs\features\com.qnx.tools.jre.win32_1.6.0.43\jre\bin.

**Comment by dbrgss** [comment]
Can you please copy the PATH variable as is.. and paste it into the issue.   

Is the PATH variable a User Variable.. or a System Variable...

**Comment by srinivas4912** [comment]
The PATH variable is a "User variable for User"
PATH : C:\bbndk_vs\features\com.qnx.tools.jre.win32_1.6.0.43\jre\bin
For the reference, below is the dialog copied from Win8.1 machine:
![image](https://f.cloud.github.com/assets/5992406/2138263/14d49b3a-9332-11e3-88a8-afd0434f240e.png)

**Comment by dbrgss** [comment]
I notice that the path has two \  Not sure if this is the cause of the grief.   Can you please edit the PATH and remove the extra \    

Also make sure there isn't any other extra \ in the rest of the path.

I am not sure if you need to restart windows for the environment variable to update or not.

**Comment by srinivas4912** [comment]
I will retest it on fresh machine again because after restart the system i didn't get java related error.

**Comment by srinivas4912** [comment]
Verified on Win8.1/Win7  VS2013/12 using v2.0.0022 plugin and found as fixed.
When we set PATH variable with "C:\bbndk_vs\features\com.qnx.tools.jre.win32_1.6.0.43\jre\bin;", no notification is shown.

**Comment by pvallabhaneni** [comment]
Issue is reproducing in the following scenario.
Preq:
System OS - Win 7 64 bit
VS Version :VS2012
Plugin Version : 2.0.0022

Scenario 1:
1. Install Visual Studio and NDK plugin on fresh machine.
2. Open Blackberry->Signing.

Actual Result: Error message (Eclipsec) is shown. This is happening only once and if error message is closed and open the settings again nor error message is shown.
Note: Environmental Variables are not modified at any of the above instances.

![capture](https://f.cloud.github.com/assets/5992422/2194414/0679946a-988b-11e3-9a50-dcba6f865a56.PNG)

Scenario 2:
1. Uninstall the plugin installed from the above steps.
2. Restart the system.
3. Install the plugin again.
4. Select Blackberry->Signing.

Actual Result: 
Bug1 : Above dialog(Eclipsec) is shown. Only after closing the dialog fr 3 times, signing dialog is shown.
Bug 2 : Observe that the path shown on the dialog is "C:\bbndk_vs\jre\bin\javaw.exe". 
Where as path set in te environment variable is: "C:\bbndk_vs\features\com.qnx.tools.jre.win32_1.6.0.43\jre\bin;"

Assigned Milestone back to "Triage".

**Comment by pvallabhaneni** [comment]
Closing this issue and created a new issue #643 .
